src/diff/props.js

```diff
@@ -35,7 +35,9 @@ let eventClock = 0;
  * @param {string} namespace Whether or not this DOM node is an SVG node or not
  */
 export function setProperty(dom, name, value, oldValue, namespace) {
-	let useCapture;
+	let useCapture,
+		split = name.split(':');
```

Comment:
The `.split()` always causes an array allocation here. I wonder if we can avoid the allocation by only checking if there even is a prefix to begin with and if so do string slicing.

```ts
let prefix = "";
// Avoid doing anything at all if there is zero chance of
// it being a prefix name.
if (name[0] === "a" || name[0] === "p") {
  let idx = name.indexOf(":")
  if (idx > -1) {
    // Only slices here, should be faster than array allocation
    prefix = name.slice(0, idx)
    name = name.slice(idx + 1)   
  }
}
```

Reply:
Yeah, good call.

I don't love the checking if first letters (we've had sone collateral damage with the "on" check) though maybe that's the best way to do it if we did want these props.

I assume Babel does support leading symbols in the JSX props (haven't checked) so perhaps following Lit with `.foo` and what not is easier for us.

Reply:
yeah in this case the first letter check is fine I think since there are not many DOM attributes starting with `a` or `p`. It's only used as an early bail out in this case here. Having a unique first character would definitely be a better choice from a detection perspective. We could leave default as attribute and then you add some prefix like a dot or `@` to mark it as a prop.

Reply:
> We could leave default as attribute and then you add some prefix like a dot or @ to mark it as a prop.

If we want this to be purely additive, and I think we do, we'll need symbols/directives for both. If someone wants to set the `value` attribute instead of the prop (as problematic as this is), they'll need some way to express this as our current setup doesn't allow it.